src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2397,6 +2397,91 @@ def load_key_and_certificates_from_pkcs12(self, data, password):
 
         return (key, cert, additional_certificates)
 
+    def generate_pkcs12(self, password, name, key, cert, ca, key_encryption,
+                        cert_encryption):
+        if password is not None:
+            utils._check_byteslike("password", password)
+        if name is not None:
+            utils._check_byteslike("name", name)
+        if key is not None and not isinstance(
+            key, (rsa.RSAPrivateKeyWithSerialization,
+                  dsa.DSAPrivateKeyWithSerialization,
+                  ec.EllipticCurvePrivateKeyWithSerialization)):
+            raise TypeError(
+                "Key must be rsa, dsa, ec, ed25519 or ed448 private key."
+            )
+        if cert is not None and not isinstance(cert, x509.Certificate):
+            raise TypeError("cert must be a certificate")
+
+        if key is None:
+            nid_key = 0
+        else:
+            assert key_encryption is not None, "No key encryption provided"
```

Comment:
The caller conrols, so it should be a real check, not an `assert`. But really you can remove this line, it's covered by the next `isinstance` check.